provider/core/src/dynutil.rs

```diff
@@ -189,6 +204,52 @@ macro_rules! impl_dyn_provider {
             $crate::serde::SerializeMarker
         );
     };
+    ($provider:ty, [ $($struct_m:ty),+, ], SERDE_SE, impl DataConverter) => {
```

Comment:
This macro is getting massive. Now that we have a `datagen` feature we could split it into what it was before #1595, and a second macro (`impl_datagen_provider`) that additionally implements `IterableDynProvider` and `DataConverter` (and which we can also put behind the `datagen` feature).

Reply:
I'm waiting for the ResourceProvider stuff to happen before making further changes here

Reply:
What ResourceProvider stuff?